src/app/containers/SignupContainer/SignupContainer.js

```diff
@@ -25,26 +25,34 @@ const SignupState = {
 };
 
 // TODO: Flexible urls and plans for reuse between project
-const SignupContainer = ({ history, onLogin, stopRedirect }) => {
-    useEffect(() => {
-        document.title = c('Title').t`Sign up - ProtonVPN`;
-    }, []);
-
+const SignupContainer = ({ match, history, onLogin, stopRedirect }) => {
     const searchParams = new URLSearchParams(history.location.search);
     const preSelectedPlan = searchParams.get('plan');
     const redirectToMobile = searchParams.get('from') === 'mobile';
     const availablePlans = checkCookie('offer', 'bestdeal') ? BEST_DEAL_PLANS : VPN_PLANS;
 
-    const [signupState, setSignupState] = useState(preSelectedPlan ? SignupState.Account : SignupState.Plan);
+    useEffect(() => {
+        document.title = c('Title').t`Sign up - ProtonVPN`;
+        // Always start at plans, or account if plan is preselected
+        if (preSelectedPlan) {
+            history.replace(`/signup/${SignupState.Account}`);
+        } else {
+            history.replace('/signup');
+        }
+    }, []);
+
+    const signupState = match.params.step;
     const [upsellDone, setUpsellDone] = useState(false);
     const [creatingAccount, withCreateLoading] = useLoading(false);
     const historyState = history.location.state || {};
     const invite = historyState.invite;
     const coupon = historyState.coupon;
 
+    const goToStep = (step) => history.push(`/signup/${step}`);
+
     const handleLogin = (...args) => {
         if (redirectToMobile) {
-            return setSignupState(SignupState.MobileRedirection);
+            goToStep(SignupState.MobileRedirection);
```

Comment:
the `return` should still be necessary here?

Reply:
oh shit yeah.. I got confused, but it's to interrupt the flow.